sdk/spring/azure-spring-cloud-autoconfigure/src/main/java/com/azure/spring/autoconfigure/storage/resource/AzureStorageProtocolResolver.java

```diff
@@ -44,27 +43,21 @@ public void setResourceLoader(ResourceLoader resourceLoader) {
     @Override
     public Resource resolve(String location, ResourceLoader resourceLoader) {
         if (AzureStorageUtils.isAzureStorageResource(location, StorageType.BLOB)) {
-            return new BlobStorageResource(getBlobServiceClientBuilder().buildClient(), location, true);
+            if (blobServiceClient != null) {
```

Comment:
okay, now i see there is a null check, but why don't we have a resolver to do one thing only to make it more follow single-responsibility principle.

for example:

class AzureStorageBlobProtocolResolver

then we can have a conditionalOnBean(blobServiceClient)

Reply:
refactored.